frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -189,15 +189,15 @@ pub fn expand_outer_inherent(
 					};
 
 					if !is_inherent {
-						first_signed_observed = true;
+						first_signed_observed = i as u32;
```

Comment:
The trait docs say 
```
On success, return the index of the first non-inherent (counting from 0).
```
but won't this return the index of the last non-inherent because we update ever time we see a signed xt?
To me it looks like this will return the index of the _last_ signed xt.

Reply:
If you change `first_signed_observed` to an `Option` you could do something to the affect of
```rust
if !is_inherent && first_signed_observed.is_none() { 
    first_signed_observed = Some(i as u32) 
}
```